src/main/java/duke/dukehelper/Parser.java

```diff
@@ -0,0 +1,135 @@
+    /**
+     * Process input string, tokenising it by a space seperator and removing trailing spaces per token.
+     * @param rawInput Input string from users
+     * @return A packet instance with different categories of information sorted out.
+     */
+    public static Packet parseInput(String rawInput){
+        String regex = "(\\s+)(/\\w+)(\\s+)";
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = null;
+        // A String buffer is used as placeholder for extracted string fragments from parsing input string.
+        String[] buffer = new String[]{"", "", "", ""};
+        Packet packet = null;
+        String input = rawInput + " /end ";
+        // Extract command type
+        buffer[0] = input.substring(0, input.indexOf(" ")).toLowerCase();
+        // Extract task name
+        buffer[1] = (input.substring(input.indexOf(" "), input.indexOf(" /", 1))).replace(" ", " ").trim();
+
+        // Initialize packet with known parameters: command and command payload
+        packet = new Packet(buffer[0], rawInput);
+        packet.setPacketPayload(buffer[1]);
+//        System.out.println(String.format("0: %s, 1: %s", buffer[0], buffer[1]));
+
+        // Implemented in case future commands with multiple param types are to be implemented
+        do {
+            input = input.substring(input.indexOf(" /")).trim();
+            if (input.equals("/end")) break;
+            // Extract param type
+            buffer[2] = input.substring(input.indexOf("/"), input.indexOf(" ")).toLowerCase();
+            // Remove param type from input string
+            input = (input.replaceFirst("(/\\w+)(\\s+)", "\0")).trim();
+            input = " ".concat(input).concat(" ");
+            // Extract param name
+            matcher = pattern.matcher(input);
+            matcher.find();
+            buffer[3] = (input.substring(0, matcher.start())).trim();
+            packet.addParamToMap(buffer[2], buffer[3]);
+        } while (true);
+
+        return packet;
+    }
+
+    /**
+     * Parses raw date and time input from the user and return a formatted string that can be parsed by DateTine class.
+     * @param input
+     * @return Formatted String in YYYY-MM-DDTHH:MM:SS
+     */
+    public static String parseRawDateTime(String input){
+        int matchCount = 0;
+        // If user uses a string token as a separator between two datetimes, say "to", remove from the string.
+        String[] tokens = input.replaceAll("[\\s]+[\\D]+[\\s]+|,", " ").split("[\\s]+");
+
+        // If input matches the required pattern, no need to continue processing
+        for (String token : tokens){
+            if (token.matches("[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}")) {
+                matchCount++;
+            }
+        }
+        if (matchCount == tokens.length) {
+            return input;
+        }
+
+        String[] output = new String[2];
+        String date = "";
+
+        if (tokens.length == 2) {
```

Comment:
Magic numbers are used quite a lot here. Given that it is a parser, it can be forgiven if it parses a lot of variable inputs, resulting in a lot of different numbers of tokens. However, since there are only a small number of options for the number of tokens, perhaps you may want to consider define some constants for the numbers?